julia-mode.el

```diff
@@ -110,8 +110,9 @@
             (syntax whitespace)
             bol)
         (group "'")
-        (or (repeat 0 8 (not (any "'"))) (not (any "\\"))
-            "\\\\")
+        (or (not (any "\\'"))
+	    (seq "\\" (or (repeat 1 9 (not (any "\\'")))
+			  (any "\\'"))))
```

Comment:
Does this match [e.g. `'\n'`, `\0`, `\x7f`, `\177`](https://docs.julialang.org/en/v1/manual/strings/#man-characters)? It doesn't seem like it would. Probably worth adding a test for those.

Reply:
They all fit into the "(1 to 9 (not (backslash or singlequote))) after backslash" clause.  I could be more prescriptive, if you like.

Reply:
That makes sense. This is fine. If you wanted to, you could add a couple comments with more explanation of the different forms of char literals and how they're matched, but I don't think that should block merging this.

Reply:
Yes, explanations would be great if @recri has the time. Also, indentation of `(seq ...` looks a bit off.